manifests/process_exporter.pp

```diff
@@ -82,12 +83,18 @@
     default => undef,
   }
 
+  if $hash_watched_processes.empty() {
+    $config_path_content = template('prometheus/process-exporter.yaml.erb')
```

Comment:
you didn't adapt the template, and the only variable reference there is `$watched_processes`

Reply:
if the hash is empty then the template is used (with the array). I've done that to keep backward compatibility so that users using this class doesn't have to change any working configuration and added a new param that can be used to setup the config_path file with more flexibility. In order to work the hash should start by `process_names:`